source/server/backtrace.h

```diff
@@ -53,80 +40,50 @@ class BackwardsTrace : Logger::Loggable<Logger::Id::backtrace> {
+                                       /* min_dropped_frames = */ nullptr);
+  }
 
   /**
    * Log the stack trace.
    */
   void logTrace() {
-    backward::TraceResolver resolver;
-    resolver.load_stacktrace(stack_trace_);
-    // If there's nothing in the captured trace we cannot do anything.
-    // The size must be at least two for useful info - there is a sentinel frame
-    // at the end that we ignore.
-    if (stack_trace_.size() < 2) {
-      ENVOY_LOG(critical, "Back trace attempt failed");
-      return;
-    }
-
-    const auto thread_id = stack_trace_.thread_id();
-    backward::ResolvedTrace first_frame_trace = resolver.resolve(stack_trace_[0]);
-    auto obj_name = first_frame_trace.object_filename;
-
-#ifdef __APPLE__
-    // The stack_decode.py script uses addr2line which isn't readily available and doesn't seem to
-    // work when installed.
-    ENVOY_LOG(critical, "Backtrace thr<{}> obj<{}>:", thread_id, obj_name);
-#else
-    char out[200];
-    ENVOY_LOG(critical,
-              "Backtrace thr<{}> obj<{}> (If unsymbolized, use tools/stack_decode.py):", thread_id,
-              obj_name);
-#endif
-
-    // Backtrace gets tagged by ASAN when we try the object name resolution for the last
-    // frame on stack, so skip the last one. It has no useful info anyway.
-
-    for (unsigned int i = 0; i < stack_trace_.size() - 1; ++i) {
-      backward::ResolvedTrace trace = resolver.resolve(stack_trace_[i]);
-      if (trace.object_filename != obj_name) {
-        obj_name = trace.object_filename;
-        ENVOY_LOG(critical, "thr<{}> obj<{}>", thread_id, obj_name);
-      }
+    ENVOY_LOG(critical, "Backtrace:");
 
-#ifdef __APPLE__
-      // In the absence of stack_decode.py, print the function name.
-      ENVOY_LOG(critical, "thr<{}> #{} {} {}", thread_id, stack_trace_[i].idx, stack_trace_[i].addr,
-                trace.object_function);
-#else
-      if (absl::Symbolize(stack_trace_[i].addr, out, sizeof(out))) {
-        ENVOY_LOG(critical, "thr<{}> #{} {} {}", thread_id, stack_trace_[i].idx,
-                  stack_trace_[i].addr, out);
+    for (int i = 0; i < stack_depth_; ++i) {
+      char out[1024];
+      const bool success = absl::Symbolize(stack_trace_[i], out, sizeof(out));
+      if (success) {
+        ENVOY_LOG(critical, "#{}: {}", i, out);
       } else {
-        ENVOY_LOG(critical, "thr<{}> #{} {} (unknown)", thread_id, stack_trace_[i].idx,
-                  stack_trace_[i].addr);
+        ENVOY_LOG(critical, "#{}: {}", i, stack_trace_[i]);
       }
-#endif
     }
-    ENVOY_LOG(critical, "end backtrace thread {}", stack_trace_.thread_id());
   }
 
   void logFault(const char* signame, const void* addr) {
     ENVOY_LOG(critical, "Caught {}, suspect faulting address {}", signame, addr);
   }
 
 private:
-  static const int MAX_STACK_DEPTH = 64;
-  backward::StackTrace stack_trace_;
+  static constexpr int MAX_STACK_DEPTH = 64;
```

Comment:
Nit: prefer moving to  the newer Envoy convention of `MaxStackDepth`  while mucking around here.

Reply:
Good idea, done.